migrations/02_singleton_libs.js

```diff
@@ -0,0 +1,140 @@
+const ethUtils = require('ethereumjs-util')
+const fs = require('fs')
+const utils = require('../utils/utils.js')
+
+const SingletonFactory = artifacts.require("SingletonFactory")
+
+module.exports = async function (deployer, network, accounts) {
+
+  let addresses = require('./addresses.json')
+  const singletons = require('./singletons.json') 
+
+  const from = singletons.from
+    ? singletons.from
+    : deployer.networks[network].from
+      ? deployer.networks[network].from
+      : accounts[0]
+
+  // Generate SingletonFactory deployment transaction:
+  const res = utils.generateDeployTx(
+      SingletonFactory.toJSON(),
+      singletons.sender.r,
+      singletons.sender.s,
+      singletons.sender.gasprice,
+      singletons.sender.gas
+    )
+
+  const deployedCode = await web3.eth.getCode(res.contractAddr)
+
+  if (deployedCode.length <= 3) {
```

Comment:
Just out of curiosity, why `<= 3`? Is it related to `0x` and `0x0` later?

Reply:
Indeed, if no code is deployed at specified addres, web3.eth.getCode actually returns '0x'. However, on the "if" condition we don't take for granted this behaviour. The minimal deployed code would occupy 1 byte, that's why '<=3' is being checked.

Reply:
Interesting. Thanks for sharing those insights!